### PR TITLE
Internal: Unwanted text in the Maps widget [ED-4782]

### PR DESCRIPTION
**Pull request by YitzhakAvm**

### CODE REVIEW

**Review by iNewLegend — APPROVED** [review]

File: includes/widgets/google-maps.php [hunk 1]
```diff
@@ -108,14 +108,26 @@ protected function register_controls() {
...
+
 				$this->add_control(
 					'api_key_notification',
 					[
 						'type' => Controls_Manager::RAW_HTML,
 						'raw' => sprintf(
-							esc_html__( 'Set your Google Maps API Key in Elementor\'s <a href="%1$s" target="_blank">Integrations Settings</a> page. Create your key <a href="%2$s" target="_blank">here.', 'elementor' ),
-							Settings::get_url() . '#tab-integrations',
-							'https://developers.google.com/maps/documentation/embed/get-api-key'
+							esc_html__( 'Set your Google Maps API Key in Elementor\'s %1$s page. Create your key %2$s', 'elementor' ),
```
Comment (matipojo) [inline includes/widgets/google-maps.php:128]: It's better to make translatable text in one paragraph and not separate them into many parts, please look at [this example](https://github.com/elementor/elementor/blob/6d61615d9b4cc9a5d659da19ab9cdd5bd6ca9f35/core/experiments/manager.php#L424) and add also the "Translators" comment

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/widgets/google-maps.php (hunk 1 above)
Comment (matipojo) [reply includes/widgets/google-maps.php]: @ItzhakAvm Please fix like the example above (with the "translators" comment, and the HTML as a translation parameter)

File: includes/widgets/google-maps.php [hunk 2]
```diff
@@ -112,8 +112,9 @@ protected function register_controls() {
 					'api_key_notification',
 					[
 						'type' => Controls_Manager::RAW_HTML,
+						/* translators: %1$s Integration settings link open tag, %2$s: Link close tag, %3$s Create API key link open tag, %4$s: Link close tag. */
 						'raw' => sprintf(
-							esc_html__( 'Set your Google Maps API Key in Elementor\'s %1$sIntegrations Settings%2$s page. Create your key %3$shere.%4$s', 'elementor' ),
+							esc_html__( 'Set your Google Maps API Key in Elementor\'s %1$sIntegrations Settings%3$s page. Create your key %3$shere.%3$s', 'elementor' ),
```
Comment (matipojo) [inline includes/widgets/google-maps.php:117]:
```suggestion
							esc_html__( 'Set your Google Maps API Key in Elementor\'s %1$sIntegrations Settings%2$s page. Create your key %3$shere.%2$s', 'elementor' ),
```

File: includes/widgets/google-maps.php [hunk 3]
```diff
@@ -112,8 +112,9 @@ protected function register_controls() {
 					'api_key_notification',
 					[
 						'type' => Controls_Manager::RAW_HTML,
+						/* translators: %1$s Integration settings link open tag, %2$s: Link close tag, %3$s Create API key link open tag, %4$s: Link close tag. */
```
Comment (matipojo) [inline includes/widgets/google-maps.php:115]:
```suggestion
						/* translators: %1$s Integration settings link open tag, %2$s: Link close tag, %3$s Create API key link open tag. */
```

You can use the close tag (the second parameter) also for closing the second link. please remove the 4th parameter (line 121)

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]
Please fix the Lint

**Comment by Aviad-Herman** [comment]
Hey @matipojo can you please retarget to 3.5.0?

File: includes/widgets/google-maps.php [hunk 4]
```diff
@@ -112,10 +112,12 @@ protected function register_controls() {
 					'api_key_notification',
 					[
 						'type' => Controls_Manager::RAW_HTML,
+						/* translators: %1$s: Integration settings link open tag, %2$s: Create API key link open tag, %3$s: Link close tag. */
```
Comment (matipojo) [inline includes/widgets/google-maps.php:115]: The translation comment should be one line above the translation.
```suggestion
```

File: includes/widgets/google-maps.php [hunk 5]
```diff
@@ -112,10 +112,12 @@ protected function register_controls() {
...
 						'raw' => sprintf(
-							esc_html__( 'Set your Google Maps API Key in Elementor\'s <a href="%1$s" target="_blank">Integrations Settings</a> page. Create your key <a href="%2$s" target="_blank">here.', 'elementor' ),
-							Settings::get_url() . '#tab-integrations',
-							'https://developers.google.com/maps/documentation/embed/get-api-key'
+							esc_html__( 'Set your Google Maps API Key in Elementor\'s %1$sIntegrations Settings%3$s page. Create your key %2$shere.%3$s', 'elementor' ),
```
Comment (matipojo) [inline includes/widgets/google-maps.php:117]: Please use `1:` & `2:` in the translation comment.
```suggestion
							/* translators: 1: Integration settings link open tag, 2: Create API key link open tag, %3$s: Link close tag. */
							esc_html__( 'Set your Google Maps API Key in Elementor\'s %1$sIntegrations Settings%3$s page. Create your key %2$shere.%3$s', 'elementor' ),
```

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by rodolphebertozzo — CHANGES_REQUESTED** [review]
Pull from 3.5.0

**Pull request merged**